### ISSUE TITLE
Inquiry about your research paper

### ISSUE THREAD

**Issue by deux1203** [imperial-qore/TranAD#67]
Hi, I think your recent paper is very impressive.
I sent these questions below by e-mail, but I didn't receive your reply. So I would like to ask again using the gitbub issue page.

Question #1. Could you please let me know why the code below should be implemented?
                      I think that I don't know the ground truth labels when I run this algorithm at the online inference phase.
                      Therefore, I guess that the code below cannot know the ground truth labels at the inference phase.
                      (the code below need the ground truth labels "actual[i]")

++++++++++++++++++++ code (@pot.py) ++++++++++++++++++++
def adjust_predicts(score, label,
                    threshold=None,
                    pred=None,
                    calc_latency=False):
  
    for i in range(len(score)):
        if actual[i] and predict[i] and not anomaly_state:
                anomaly_state = True
                anomaly_count += 1
                for j in range(i, 0, -1):
                    if not actual[j]:
                        break
                    else:
                        if not predict[j]:
                            predict[j] = True
                            latency += 1
        elif not actual[i]:
            anomaly_state = False
        if anomaly_state:
            predict[i] = True
++++++++++++++++++++ code (@pot.py) ++++++++++++++++++++

Question #2. Could you please let me know where the validation code is?
                      I read that you divided the training time-series into 80% training data and 20% validation data.
                      But I cannot find the validation parts in your code.
                      (I wonder that I can use the validation data labels because I guess that the TranAD model is a kind of unsupervised learning.)

I look forward to hearing from you.
Thank you.

**Comment by shreshthtuli** [comment]
Thanks for your questions.
1. Please refer to the Omnianomaly paper and code repo for more details on why this function is used.
2. Please refer to process.py.
